tidy3d/plugins/adjoint/components/medium.py

```diff
@@ -206,6 +241,24 @@ class JaxCustomMedium(CustomMedium, JaxObject):
         jax_field=True,
     )
 
+    @pd.validator("eps_dataset", always=True)
+    def _is_not_3d(cls, val):
+        """Ensure the custom medium pixels contain at least one dimension with only pixel thick."""
+
+        for field_dim in "xyz":
+            field_name = f"eps_{field_dim}{field_dim}"
+            data_array = val.field_components[field_name]
+            coord_lens = [len(data_array.coords[key]) for key in "xyz"]
+            dims_len1 = [val == 1 for val in coord_lens]
+            if sum(dims_len1) == 0:
+                raise SetupError(
+                    "For adjoint plugin, 'JaxCustomMedium' can't contain more than "
+                    "1 pixel along all three dimensions. "
+                    f"Detected 3D pixelated grid in '{field_name}' component of 'eps_dataset.'."
```

Comment:
Message is a bit confusing, namely "more than 1 pixel along all three dimensions". Rather something like `"'JaxCustomMedium' cannot be 3D, needs to be single pixel along at least one dimension."`? Also there are two periods at the end of the message, `'eps_dataset.'.`

Reply:
3D is a bit confusing I think because the geometry itself can be 3D, it just can't have more than 1 pixel along each dimension. I guess maybe another way of saying this is: 
"For adjoint plugin, the 'JaxCustomMedium' is restricted to a 1D or 2D pixellated grid. It may not contain multiple pixels along all 3 dimensions. Detected 3D pixelated grid in '{field_name}' component of 'eps_dataset'."

How is that? (also resolves the double period thing)

Reply:
Sounds good.